code/modules/mob/living/living_defense.dm

```diff
@@ -60,6 +60,9 @@
 //this proc handles being hit by a thrown atom
 /mob/living/hitby(atom/movable/AM, speed = 5)
 	. = TRUE
+	if(SEND_SIGNAL(src, COMSIG_LIVING_PRE_THROW_IMPACT, AM) & COMPONENT_PRE_THROW_IMPACT_DODGED)
+		return FALSE
```

Comment:
This proc explicitly doesn't call parent in many cases. I feel like this probably should be in throw_impact()